code/datums/cargo.dm

```diff
@@ -0,0 +1,381 @@
+	data["name"] = name
+	data["description"] = description
+	data["tag_line"] = tag_line
+	data["shuttle_time"] = shuttle_time
+	data["shuttle_price"] = shuttle_price
+	data["available"] = available
+	data["price_modifier"] = price_modifier
+	return data
+
+/*
+	A category displayed in the cargo order app
+*/
+/datum/cargo_category
+	var/name = "cargo_category" //Name of the category
+	var/display_name = "Cargo Category"
+	var/description = "You should not see this" //Description of the Category
+	var/icon = "gear" //NanoUI Icon for the category
+	var/price_modifier = 1 //Price Modifier for the category
+	var/list/items = list() //List of items in the category
+
+// Gets a list of the cargo category - to be json encoded
+/datum/cargo_category/proc/get_list()
+	var/list/data = list()
+	data["name"] = name
+	data["display_name"] = display_name
+	data["description"] = description
+	data["icon"] = icon
+	data["price_modifier"] = price_modifier
+	return data
+
+// Gets a list of the items in the cargo category - to be json encoded
+/datum/cargo_category/proc/get_item_list()
+	var/list/item_list = list()
+	for(var/datum/cargo_item/ci in items)
+		item_list.Add(list(ci.get_list()))
+	return item_list
+
+/*
+	A order placed in the cargo order app.
+	Contains multiple order items
+*/
+/datum/cargo_order
+	var/list/items = list() //List of cargo_order_items in the order
+	var/order_id = 0 //ID of the order
+	var/price = 0 //Total price of the order
+	var/item_id = 1 //Current id of the item in the order
+	var/item_num = 0 //Numer of items in the container - Used to limit the items in the crate
+	var/status = "basket" //Status of the order: basket - Adding items, submitted - Submitted to cargo, approved - Order sent to suppliers, rejected - Order has been denied, shipped - Has been shipped to the station, delivered - Order has been delivered
+	var/container_type = "" //Type of the container for the order - cate, box
+	var/list/required_access = list() //Access required to unlock the crate
+	var/can_add_items = 1 //If new items can be added to the order 
+	var/customer = null //Person that ordered the items
+	var/authorized_by = null //Person that authorized the order
+	var/received_by = null //Person the order has been delivered to by cargo
+	var/time_submitted = null //Time the order has been sent to cargo
+	var/time_approved = null //Time the order has been approved by cargo
+	var/time_shipped = null //Time the order has been shipped to the station
+	var/time_delivered = null //Time the order has been delivered
+	var/tracking_code = null //Use this code with the order ID to get details about the order
+	var/partial_shipment_fee = 0 //Partial Shipment Fee for the order
+	var/reason = null //Reason for the order
+
+//Gets the tracking code for the order. Generates one if it does not exist already
+/datum/cargo_order/proc/get_tracking_code()
+	if(!tracking_code)
+		tracking_code = "asdf"
+		//Generate Tracking code TODO: Finish That
```

Comment:
I´m thinking about letting the user enter their own code when the order is placed